Allow users to disable certain reboot checks

qmfrederik opened the pull request:
I had an issue where my Azure VM went into a reboot loop after I applied a Powershell DSC script that included xPendingReboot.

Basically, there was a stale file rename operating pending, so every time the DSC script ran, it would detect that, reboot the VM, and we got in a loop.

The case is, I did not want to check for pending file rename operations - I was only looking for a reboot that could have been triggered by a Windows Update package I installed.

So I've created this pull request that allows the user to specify for which 'reboot sources' (s)he wants to check. For example, if you want to bypass the pending file rename operations, you could just set `SkipPendingFileRename = $True`.

qmfrederik committed:
Add options to skip the individual reboot sources, such as the pending file rename

msftclas commented:
Hi __@qmfrederik__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.microsoft.com.
    </span>

TTYL, MSBOT;

msftclas commented:
__@qmfrederik__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSBOT;

qmfrederik committed:
Add [write] flag as per PR feedback

qmfrederik commented on DSCResources/MSFT_xPendingReboot/MSFT_xPendingReboot.psm1:
```diff
@@ -57,33 +57,58 @@ Function Test-TargetResource
     [Parameter(Mandatory=$true)]
-    [string]$Name
+    [string]$Name,
+    [bool]$SkipComponentBasedServicing,
```
Hi @TravisEz13 ,

Can you elaborate? By default, all the $skip\* variables will be set to $false, so the if(-not $skip*) blocks will be entered, and the code should behave like it did before this PR.

Let me know if I missed something.

qmfrederik committed:
Add prototype test

qmfrederik committed:
Add parameter documentation to the schema

qmfrederik committed:
Update README with info on the various  flags

qmfrederik committed:
Fix the parameters for Set-TargetResource

qmfrederik commented:
Hi @TravisEz13 ,

I think I addressed most of your feedback; let me know if something is missing.

Regarding the tests: I added a first test; however, it seems my mocks are not working (correctly).
I have to admit I'm now to Pester and there were no tests in the repository which I could base my test off.

Would be great if you could provide some guidance here,

Thanks.

qmfrederik committed:
Convert MOF file to ASCII encoding

qmfrederik committed:
Specify output types

qmfrederik committed:
Fix missing ]

qmfrederik committed:
Add Pester tests for Get-TargetResource, Test-TargetResource

qmfrederik commented:
Hi @TravisEz13 ,

Have a look at the updated code:
* `Test-TargetResource` is based off `Get-TargetResource`; the tests for `Test-TargetResource` now mock the values of `Get-TargetResource`
* I've added tests for `Get-TargetResource` as well

Looks like something went wrong with the mocking of `Get-WmiObject`; it would be great if you could have a look at that and let me know if there's something I missed.

Cheers

qmfrederik commented:
Hi @TravisEz13 ,

Actually, I took care of refactoring the code so that `Invoke-Method` is no longer used by the PowerShell module.

The test now fail because it seems like I can't get Pester to mock `Get-WmiObject`.

Any thoughts?

qmfrederik committed:
Fix unit tests

qmfrederik commented:
Hi @TravisEz13 ,

Cool, thanks! Looks like we're all set now?

Frederik.
